pkg/cli/cmds/agent_service_windows.go

```diff
@@ -0,0 +1,138 @@
+// +build windows
+
+package cmds
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+	"unsafe"
+
+	"github.com/rancher/k3s/pkg/cli/cmds"
+	"github.com/urfave/cli"
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+const (
+	defaultServiceDisplayName = "rke2"
+	defaultServiceDescription = "Rancher Kubernetes Engine v2 (agent) see docs https://github.com/rancher/rke2#readme"
+)
+
+func serviceSubcommand() cli.Command {
```

Comment:
Does this need to be returned from a function? Can it be an unexported package var?

Reply:
yes it could, not a bad idea